### PR TITLE
Env

### LINKED ISSUE #24: Implement env

### PR DESCRIPTION
**Pull request by FedericoPonzi**
Should fix #24 
There should be feature parity between `env` and this implementation :+1: sorry for the formatting, I had some problems with the rustfmt (it's complaining about syntax error, like edition is supposed to be a string instead of int etc).

### CODE REVIEW

File: env/src/env.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: echo
+version: "0.0.0"
+author: Federico Ponzi <federico.ponzi92@gmail.com>
+about: "
+     env [OPTION]... [-] [NAME=VALUE]... [COMMAND [ARG]...]\n\n
+    Set each NAME to VALUE in the environment and run COMMAND.:\n\n"
```
Comment (GrayJack) [inline env/src/env.yml]: About is not usage, put a text describing the utility, like 

```
"Set environment and execute command, or print environment\n\n
The env utility executes another [COMMAND] after modifying the environment as specified on the command line. Each [NAME=VALUE] option specifies the setting of an environment variable, name, with a value of value.  All such environment variables are set before the [COMMAND] is executed."
```
clap generates a good usage by default, just use if it's way worse than expected

File: env/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,96 @@
+use std::collections::HashMap;
+use std::process;
+use std::process::Command;
+use std::{env, io};
```
Comment (GrayJack) [inline env/src/main.rs]: Use nested `use` when it's the same base library

File: env/src/env.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+args:
+    - OPTIONS:
+        help: Environement variables in the form of NAME=VALUE and the COMMAND to run with its arguments.
+        multiple: true
+    - IGNORE_ENVIRONMENT:
```
Comment (GrayJack) [inline env/src/env.yml]: For flags and options, use the same name as long, I don't really mind the way you did, but all other utilities does that and consistency is good :3

File: env/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+                None => {
+                    cmd.push(word_str);
+                }
+            }
+        }
+    });
+    let mut unset_keys = Vec::new();
+    matches.values_of("REMOVE_VAR").map(|keys| {
+        keys.for_each(|k| unset_keys.push(k.to_owned()));
+    });
```
Comment (GrayJack) [inline env/src/main.rs]: I think would be better to use `if let` statement

File: env/src/main.rs [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+use std::{env, io};
+
+use clap::{load_yaml, App};
+
+fn main() {
+    let yaml = load_yaml!("env.yml");
+    let matches = App::from_yaml(yaml).get_matches();
+    let mut kv = HashMap::new();
+    let mut cmd = Vec::new();
+    matches.values_of("OPTIONS").map(|m| {
```
Comment (GrayJack) [inline env/src/main.rs]: I think would be better to use `if let` statement

File: env/src/main.rs [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+            .args(cmd)
+            .envs(&env_vars)
+            .status()
+            .expect(&format!("{} failed to start.", cmd_name));
+        println!("done spawining..");
+    } else {
+        for (key, value) in env_vars.iter() {
+            print!("{}={}", key, value);
+            if !null_eol {
+                print!("\n");
```
Comment (GrayJack) [inline env/src/main.rs]: Empty `println()` here is better

**Comment by FedericoPonzi** [comment]
Please let me know if now it's in a good shape or it needs more changes :+1:

**Comment by GrayJack** [comment]
Nice, since you implemented all features that are common between all coreutils implementations that I know (GNU, FreeBSD, OpenBSD, NetBSD, DragonflyBSD and Illumon/Solaris) I'm ready to merge, unless you want to implement more features before the merge

**Comment by FedericoPonzi** [comment]
Sorry, I don't get it if there is feature parity what kind of other
features should be implemented? :thinking:
I'm looking forward at the next program already if this is what you meant,
but I'll open subsequent PRs in that case.

Il mer 2 ott 2019, 18:19 Eric Shimizu Karbstein <notifications@github.com>
ha scritto:

**Comment by GrayJack** [comment]
> Sorry, I don't get it if there is feature parity what kind of other features should be implemented?

Not really, I was just asking in case you wanted to implement non common features, some times people want to implement it all, but the commons ones are the minimum guarantee of the projects

**Pull request merged**